### PR TITLE
feat(Auth): Handle session expiry for auth session

### PR DESCRIPTION
**Pull request by royjit**
- Handles session expiry from AWSMobileClient
- Reorganized files
- Initial setup for integration test
  - Changed AWSMobileClient to use the singleton object AWSMobileClient.default()
  - Added reset() api for the plugin

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthorizationProviderAdapter.swift [hunk 1]
```diff
@@ -37,4 +38,31 @@ class AuthorizationProviderAdapter: AuthorizationProviderBehavior {
...
+
+    private func setupListener() {
+        awsMobileClient.addUserStateListener(self) { [weak self] state, _ in
+            guard let self = self else {
+                return
+            }
+            switch state {
+            case .signedOutFederatedTokensInvalid,
+                 .signedOutUserPoolsTokenInvalid:
+                print("AWSMobileClient Event listener - \(state)")
```
Comment (palpatim) [inline AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthorizationProviderAdapter.swift:50]: Here and throughout: Use `Amplify.Logging`, not `print()`

File: AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthorizationProviderBehavior.swift [hunk 2]
```diff
@@ -13,4 +13,6 @@ protocol AuthorizationProviderBehavior {
                       completionHandler: @escaping (Result<AuthSession, AuthError>) -> Void)
 
     func invalidateCachedTemporaryCredentials()
+
+    func reset()
```
Comment (palpatim) [inline AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthorizationProviderBehavior.swift:17]: Avoid declaring `reset` directly in behavior protocols in favor of a separate `resettable` conformance on the plugin implementation. Is there a reason we have to put this in the core behavior?

File: AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthorizationProviderAdapter.swift (hunk 1 above)
Comment (royjit) [reply AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthorizationProviderAdapter.swift]: Changed

File: AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthorizationProviderBehavior.swift (hunk 2 above)
Comment (royjit) [reply AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthorizationProviderBehavior.swift]: Added Resettable conformance to the implementation.

**Pull request merged**